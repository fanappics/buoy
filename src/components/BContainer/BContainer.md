Sample BContainer usage:

```js
<b-container class="edit_me" label="Container Header" headerLevel="3">
  <b-button>Test</b-button>
  <b-button>Test2</b-button> 
</b-container>
```
