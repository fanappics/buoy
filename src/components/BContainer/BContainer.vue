<template>
  <section role="contentinfo" :aria-label="label">
    <component v-if=!hideLabel :is=headerElement() class="header">{{ label }}</component>
    <div>
      <slot></slot>
    </div>
  </section>
</template>

<script>
  /**
  * Buoy Container Component
  */
export default {
  name: 'b-container',
  props: {
    /**
    * Label text.  Will be used in display label and aria-label.
    */
    label: {
      type: String,
      required: true
    },
    /**
    * Display label importance.  Valid values are 1-6.  Will generate h1-h6 elements.
    */
    headerLevel: {
      type: [String, Number]
    },
    /**
    * Used to hide the display label.  Label prop will still be used for aria-label.
    */
    hideLabel: {
      type: Boolean
    }
  },
  created: function() {
    if (typeof this.headerLevel === "undefined") {
      console.warn ('No headerLevel property set, defaulting to 2')
    }
    if (isNaN(this.headerLevel) || this.headerLevel < 1 || this.headerLevel > 6) {
      console.error('Invalid headerLevel property set. Valid input are numbers 1-6.  Reverting to default level 2.')
    }
  },
  data: function() {
    return {
      headerElement: () => {
        if (isNaN(this.headerLevel) || this.headerLevel < 1 || this.headerLevel > 6) {
          return `h2`
        } else {
          return `h${this.headerLevel}`
        }
      }
    }
  }
}
</script>

<style scoped>
  .header {
    margin: 0.125rem 0.3125rem;
  }
  div {
    display: flex;
    flex-grow: 1;
    border:1px solid rgb(217,217,217);
    border-radius: 0.1875rem;
    margin: 0.125rem 0.3125rem;
    padding: 0.625rem;
  }
  section {
    display: flex;
    flex: 1;
    flex-direction: column;
  }
</style>
<docs>
```js
<b-container class="edit_me" label="Container Header" headerLevel="3">
  <b-button>Test</b-button>
  <b-button>Test2</b-button> 
</b-container>
```

</docs>