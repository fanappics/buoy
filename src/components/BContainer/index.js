import BContainer from './BContainer.vue'

/* istanbul ignore next */
BContainer.install = function install (Vue) {
  Vue.component(BContainer.name, BContainer)
}

export default BContainer
