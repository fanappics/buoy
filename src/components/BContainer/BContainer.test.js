import { mount } from 'vue-test-utils'
import BContainer from './BContainer'

describe('BContainer', () => {
  it('renders correctly', () => {
    const label = 'hello world'
    const headerLevel = 1
    const wrapper = mount(BContainer, {
      propsData: { label, headerLevel }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it(`renders an element with no h element if hideLabel is passed`, () => {
    const label = 'hello world'
    const hideLabel = true
    const wrapper = mount(BContainer, {
      propsData: { label, hideLabel }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it(`renders props.label in header element when passed.  Can specify header level with prop`, () => {
    const label = 'hello world'
    const headerLevel = 6
    const wrapper = mount(BContainer, {
      propsData: { label, headerLevel }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it(`Default to H2 if label is passed in without a headerLevel property.`, () => {
    console.warn = jest.fn()
    const label = 'hello world'
    const wrapper = mount(BContainer, { propsData: { label } })
    expect(wrapper.html()).toMatchSnapshot()
    expect(console.warn).toHaveBeenCalled()
  })
  it(`Default to H2 if label is passed in with an invalid headerLevel property. Should log to console.error.`, () => {
    console.error = jest.fn()
    const label = 'hello world'
    const headerLevel = 'The Greatest Importance'
    const wrapper = mount(BContainer, { propsData: { label, headerLevel } })
    expect(wrapper.html()).toMatchSnapshot()
    expect(console.error).toHaveBeenCalled()
  })
})
