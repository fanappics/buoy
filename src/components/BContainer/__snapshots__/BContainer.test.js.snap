// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BContainer Default to H2 if label is passed in with an invalid headerLevel property. Should log to console.error. 1`] = `
<section role="contentinfo" aria-label="hello world">
    <div>
        <!---->
        <h2 class="header">hello world</h2>
    </div>
    <div class="bordered" name="fade"></div>
</section>
`;

exports[`BContainer Default to H2 if label is passed in without a headerLevel property. 1`] = `
<section role="contentinfo" aria-label="hello world">
    <div>
        <!---->
        <h2 class="header">hello world</h2>
    </div>
    <div class="bordered" name="fade"></div>
</section>
`;

exports[`BContainer renders an element with no h element if hideLabel is passed 1`] = `
<section role="contentinfo" aria-label="hello world">
    <div>
        <!---->
        <!---->
    </div>
    <div class="bordered" name="fade"></div>
</section>
`;

exports[`BContainer renders correctly 1`] = `
<section role="contentinfo" aria-label="hello world">
    <div>
        <!---->
        <h1 class="header">hello world</h1>
    </div>
    <div class="bordered" name="fade"></div>
</section>
`;

exports[`BContainer renders props.label in header element when passed.  Can specify header level with prop 1`] = `
<section role="contentinfo" aria-label="hello world">
    <div>
        <!---->
        <h6 class="header">hello world</h6>
    </div>
    <div class="bordered" name="fade"></div>
</section>
`;
